bdai_ros2_wrappers/bdai_ros2_wrappers/subscription.py

```diff
@@ -78,3 +81,137 @@ def wait_for_message(
         future.cancel()
         return None
     return future.result()
+
+
+def wait_for_messages(
+    node: rclpy.node.Node,
+    topics: typing.List,
+    mtypes: typing.List,
+    **kwargs: typing.Any,
+) -> typing.Any:
+    """Waits for messages to arrive at multiple topics within a given time window.
+
+    Uses message_filters.ApproximateTimeSynchronizer. This function blocks
+    until receiving the messages or when a given timeout expires. Assumes the
+    given node is spinning by some external executor.
+
+    Requires the user to pass in a node, since
+    message_filters.Subscriber requires a node upon construction.
+
+    Args:
+        node (Node): the node being attached
+        topics (list): List of topics
+        mtypes (list): List of message types, one for each topic.
+        kwargs: additional arguments, including
+            delay  (float) The delay in seconds for which the messages
+                could be synchronized.
+            allow_headerless (bool): Whether it's ok for there to be
+                no header in the messages.
+            sleep (float) the amount of time to wait before checking
+                whether messages are received
+            timeout (float or None): Time in seconds to wait. None if forever.
+                If exceeded timeout, self.messages will contain None for
+                each topic.
+            latched_topics (set): a set of topics for which the publisher latches (i.e.
+                sets QoS durability to transient_local).
+
+    """
+    return _WaitForMessages(node, topics, mtypes, **kwargs).messages
+
+
+class _WaitForMessages:
+    def __init__(
+        self,
+        node: rclpy.node.Node,
+        topics: typing.List,
+        mtypes: typing.List,
+        queue_size: int = 10,
+        delay: float = 0.2,
+        allow_headerless: bool = False,
+        sleep: float = 0.5,
+        timeout: typing.Optional[float] = None,
+        verbose: bool = False,
+        exception_on_timeout: bool = False,
+        latched_topics: typing.Optional[typing.Set] = None,
```

Comment:
@kaiyu-zheng meta: there is no such thing as a latched topic in ROS 2. Consider allowing the user to specify QoS policies for each topic. Perhaps a list of (type, name, qos) triplets, like `create_subscription` takes?

Reply:
Once the set of arguments gets reduced, you can just forward `**kwargs` to `message_filters.ApproximateTimeSynchronizer`.

Reply:
I thought it may be a convenient feature to provide a default QoS profile setting for the latching use case (although latch is not a concept in ROS2). But I understand this is not standard practice for ROS2. Can make the change.

Reply:
Addressed the first comment.

Regarding `**kwargs`, the kwargs for `message_filters.Subscriber` and `message_filters.ApproximateTimeSynchronizer` are different, so I can't pass `**kwargs` into `message_filters.ApproximateTimeSynchronizer`'s constructor. Lmk if I'm missing something.